jerry-core/ecma/builtin-objects/ecma-builtin-helpers-date.cpp

```diff
@@ -1172,39 +1146,143 @@ ecma_date_value_to_utc_string (ecma_number_t datetime_number) /**< datetime */
-  ecma_date_insert_num_with_sep (&output_str_p, day, LIT_MAGIC_STRING_TIME_SEP_U, 2);
+  ecma_string_t *date_string_p = ecma_new_ecma_string_from_utf8 (character_buffer,
+                                                                 result_string_length);
+
+  return ecma_make_normal_completion_value (ecma_make_string_value (date_string_p));
+} /* ecma_date_value_to_iso_string */
+
+/**
+ * Common function to create a date string from a numeric value.
+ *
+ * Used by:
+ *        - The Date.prototype.toDateString routine.
+ *
+ * @return completion value
+ *         Returned value must be freed with ecma_free_completion_value.
+ */
+ecma_completion_value_t
+ecma_date_value_to_date_string (ecma_number_t datetime_number) /**<datetime */
+{
+  /*
+   * Character length of the result string.
+   */
+  constexpr uint32_t result_string_length = 10;
+
+  lit_utf8_byte_t character_buffer[result_string_length];
+  lit_utf8_byte_t *dest_p = character_buffer;
+
+  ecma_number_t year = ecma_date_year_from_time (datetime_number);
+  dest_p = ecma_date_value_number_to_bytes (dest_p, (int32_t) year, 4);
+  *dest_p++ = LIT_CHAR_MINUS;
 
   /*
    * Note:
    *      'ecma_date_month_from_time' (ECMA 262 v5, 15.9.1.4) returns a number from 0 to 11,
    *      but we have to print the month from 1 to 12 for ISO 8601 standard (ECMA 262 v5, 15.9.1.15).
    */
-  ecma_number_t month = ecma_date_month_from_time (datetime_num) + 1;
-  ecma_date_insert_num_with_sep (&output_str_p, month, LIT_MAGIC_STRING_MINUS_CHAR, 2);
+  ecma_number_t month = ecma_date_month_from_time (datetime_number) + 1;
+  dest_p = ecma_date_value_number_to_bytes (dest_p, (int32_t) month, 2);
+  *dest_p++ = LIT_CHAR_MINUS;
 
-  ecma_number_t year = ecma_date_year_from_time (datetime_num);
-  ecma_date_insert_num_with_sep (&output_str_p, year, LIT_MAGIC_STRING_MINUS_CHAR, 4);
+  ecma_number_t day = ecma_date_date_from_time (datetime_number);
+  dest_p = ecma_date_value_number_to_bytes (dest_p, (int32_t) day, 2);
 
-  return ecma_make_normal_completion_value (ecma_make_string_value (output_str_p));
-} /* ecma_date_value_to_iso_string */
+  JERRY_ASSERT (dest_p - character_buffer == result_string_length);
+
+  ecma_string_t *date_string_p = ecma_new_ecma_string_from_utf8 (character_buffer,
+                                                                 result_string_length);
+
+  return ecma_make_normal_completion_value (ecma_make_string_value (date_string_p));
+} /* ecma_date_value_to_date_string */
+
+/**
+ * Common function to create a time string from a numeric value.
+ *
+ * Used by:
+ *        - The Date.prototype.toTimeString routine.
+ *
+ * @return completion value
+ *         Returned value must be freed with ecma_free_completion_value.
+ */
+ecma_completion_value_t
+ecma_date_value_to_time_string (ecma_number_t datetime_number) /**<datetime */
+{
+  /*
+   * Character length of the result string.
+   */
+  constexpr uint32_t result_string_length = 12;
```

Comment:
Okay than, I thought we won't use C++11 features until #470 isn't closed. Apart from that the patch looks good to me.